sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/KeyClientBuilder.java

```diff
@@ -189,7 +195,7 @@ public KeyAsyncClient buildAsyncClient() {
         KeyServiceVersion serviceVersion = version != null ? version : KeyServiceVersion.getLatest();
 
         if (pipeline != null) {
-            return new KeyAsyncClient(vaultUrl, pipeline, serviceVersion);
+            return new KeyClientImpl(vaultUrl.toString(), pipeline, serviceVersion);
```

Comment:
Given the change to using `String` instead of `URL` should the property in the class be changed to `String` so it doesn't need to be converted back to a String when built.